native/wasmex/src/namespace.rs

```diff
@@ -148,13 +146,12 @@ pub fn receive_callback_result<'a>(env: Env<'a>, args: &[Term<'a>]) -> Result<Te
     let success = atoms::success() == args[1];
     let results = args[2].decode::<ListIterator>()?;
 
-    //TODO: use real signature
-    let signature = std::sync::Arc::new(FuncSig::new(vec![Type::I32], vec![Type::I32]));
-    let results = match decode_function_param_terms(signature.returns(), results.collect()) {
+    let return_types = token_resource.token.2.clone();
+    let results = match decode_function_param_terms(&return_types, results.collect()) {
         Ok(v) => v,
         Err(_reason) => {
             return Err(Error::Atom(
-                "could not convert callback result param to expected return signature (`{}`)",
+                "could not convert callback result param to expected return signature",
```

Comment:
here is the error message we give back to elixir when we can't convert return values back to WASM values. Originally, I wanted to include `_reason` into the error but couldn't figure out how. Maybe you have a good idea. If not, this generic message is still better than nothing I guess.